dpctl/tensor/_reduction.py

```diff
@@ -118,7 +118,7 @@ def _reduction_over_axis(
                 dpt.full(
                     res_shape,
                     _identity,
-                    dtype=_default_reduction_type_fn(inp_dt, q),
+                    dtype=dtype,
```

Comment:
With this change, the call to ``astype`` is no longer necessary.

It also means that when ``logsumexp`` or ``reduce_hypot`` reduce over an empty axis or array (e.g., ``dpt.logsumexp(dpt.ones((1, 0, 1), dtype="i4"), axis=1, dtype="i4")``) you get ``OverflowError`` instead of silently casting the identity to the output type.

For now, the ``astype`` can be removed. I've experimented with removing this branch in #1465 too.

Reply:
Yes, and it should not be `dtype`, it should be `res_dt`.